Code/Framework/AzFramework/Tests/Spawnable/SpawnableEntitiesInterfaceTests.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzCore/std/containers/array.h>
+#include <AzFramework/Spawnable/SpawnableEntitiesInterface.h>
+#include <AzTest/AzTest.h>
+
+namespace UnitTest
+{
+    //
+    // SpawnableEntityContainerView
+    //
+
+    class SpawnableEntityContainerViewTest : public ::testing::Test
+    {
+    protected:
+        AZStd::array<AZ::Entity*, 4> m_values{ reinterpret_cast<AZ::Entity*>(1), reinterpret_cast<AZ::Entity*>(2),
+                                               reinterpret_cast<AZ::Entity*>(3), reinterpret_cast<AZ::Entity*>(4) };
```

Comment:
FYI: This could take advantage of the deduction guide to deduce array template parameters
```suggestion
        AZStd::array m_values{ reinterpret_cast<AZ::Entity*>(1), reinterpret_cast<AZ::Entity*>(2),
                                               reinterpret_cast<AZ::Entity*>(3), reinterpret_cast<AZ::Entity*>(4) };
```

Reply:
MSVC doesn't agree yet that the above is a good idea. Probably means template argument deduction hasn't been added for `AZStd::array` yet, which is out of scope for a simple commit like this.

Reply:
It actually has been added though.